### ISSUE TITLE
🚨 Potential Security Vulnerability - huntr.dev

### ISSUE THREAD

**Issue by huntr-helper** [evilsocket/stork#5]
#### ⚠️ Potential Vulnerability in stork

👋 Hello, @evilsocket - @dwisiswant0 has disclosed a potential vulnerability in your repository. To validate or invalidate this potential vulnerability, please visit https://huntr.dev/bounties/1-golang-stork and join our community in helping secure open-source code.

---

#### ☎️ Need further support?

Come and join us on our [Discord](https://huntr.dev/discord) and a member of our team will be happy to help! 🤗

cc - @JamieSlome

**Comment by evilsocket** [comment]
if you have to spam at least make sure those links can be visible without giving you an access token to my github.

**Comment by adam-nygate** [comment]
Hi @evilsocket, here's the details of the disclosure:

# :writing_hand: Description
When new version `from_user` prompted, it can execute OS command, since it use `fmt.Scanln` _(which is space-separated)_, the next value given will be executed automatically when the program finished.
# :detective: Proof of Concept
Create Stork script & save as `file.stork`:
```bash
#!/usr/bin/env stork -f
git:changelog

version:file "cmd/PACKAGE/version.go"
version:from_user

git:create_tag $VERSION
```
Run stork -f file.stork, it'll ask you to enter the latest version for your Go package. Fill in the payload as follows:
```
abc duname -sr
```
The value before space will define new version for program and the next one will be executed as a command.
![image](https://user-images.githubusercontent.com/4394852/115870472-312b2600-a437-11eb-8b48-b5d91efc1b11.png)
# :boom: Impact
An attacker could include arguments that allow unintended commands or code to be executed, allow sensitive data to be read or modified or could cause other unintended behavior through malicious version defined for the Go package.

Let me know if you think the vulnerability is valid or not and I will manually mark it via our platform.

**Comment by evilsocket** [comment]
i'm not sure i'm following, especially the impact section ... stork is made to run as unprivileged user, so what's exactly the impact? it'd be running commands you wanted to run and that you typed as part of the version? 

Also the behaviour you're referring to has been patched few versions ago with this https://github.com/evilsocket/stork/commit/4b03e376b63438cc2048b307d6b59672843ac5df ...

**Comment by adam-nygate** [comment]
> i'm not sure i'm following, especially the impact section ... stork is made to run as unprivileged user, so what's exactly the impact? it'd be running commands you wanted to run and that you typed as part of the version?
> 
> Also the behaviour you're referring to has been patched few versions ago with this [4b03e37](https://github.com/evilsocket/stork/commit/4b03e376b63438cc2048b307d6b59672843ac5df) ...

@dwisiswant0 feel free to jump in to let @evilsocket know your thoughts...

**Comment by dwisiswant0** [comment]
> > i'm not sure i'm following, especially the impact section ... stork is made to run as unprivileged user, so what's exactly the impact? it'd be running commands you wanted to run and that you typed as part of the version?
> > 
> > Also the behaviour you're referring to has been patched few versions ago with this [4b03e37](https://github.com/evilsocket/stork/commit/4b03e376b63438cc2048b307d6b59672843ac5df) ...
> 
> @dwisiswant0 feel free to jump in to let @evilsocket know your thoughts...

Yep, it's been patched — written in the submission reference. Does Huntr received submissions from own patches, @adam-nygate?

**Comment by evilsocket** [comment]
patched or not i think that the "impact" is just none

**Comment by evilsocket** [comment]
it's like saying that bash is vulnerable because it allows you to run the commands that you type

**Comment by adam-nygate** [comment]
> > > i'm not sure i'm following, especially the impact section ... stork is made to run as unprivileged user, so what's exactly the impact? it'd be running commands you wanted to run and that you typed as part of the version?
> > > Also the behaviour you're referring to has been patched few versions ago with this [4b03e37](https://github.com/evilsocket/stork/commit/4b03e376b63438cc2048b307d6b59672843ac5df) ...
> > 
> > 
> > @dwisiswant0 feel free to jump in to let @evilsocket know your thoughts...
> 
> Yep, it's been patched — written in the submission reference. Does Huntr received submissions from own patches, @adam-nygate?

Ah got it - unfortunately, we can't reward bounties for already patched vulnerabilities.

Noted on the impact, @evilsocket. Thanks for your help. I'll go ahead and mark the vulnerability as no longer valid.
